src/Pages/SignUp/SignUp.js

```diff
@@ -0,0 +1,381 @@
+import React from "react";
+import styled from "styled-components";
+import { Link } from "react-router-dom";
+import { useState, useEffect } from "react";
+import { signupAPI } from "../../config";
+
+const SignUp = () => {
+  const [userInfo, setUserInfo] = useState({
+    firstName: "",
+    lastName: "",
+    email: "",
+    userName: "",
+    password: "",
+  });
+
+  const [firstNameValid, setFirstNameValid] = useState(true);
+  const [emailValid, setEmailValid] = useState(true);
+  const [userNameValid, setUserNameValid] = useState(true);
+  const [passwordValid, setPasswordValid] = useState(true);
+
+  const handleInput = (name, e) => {
+    setUserInfo({ ...userInfo, [name]: e.target.value });
+  };
+
+  console.log("password: ", userInfo.password);
+
+  const isFirstNameValid = () => {
+    userInfo.firstName.length === 0
+      ? setFirstNameValid(false)
+      : setFirstNameValid(true);
+  };
```

Comment:
```suggestion
  const isFirstNameValid = (e) => {
    const isValid = e.target.value.length
    setFirstNameValid(!isValid)
  };
```
1. 나머지 함수들도 이런식으로 바꿔주세요
2. 하지만 그 보다도 여러 함수를 하나로 합쳐주는 걸 시도해보시면 좋겠습니다

Reply:
1. 적용했습니다 훨씬 간결하네요 감사합니다 🙏🏽
2. 하지만 함수를 합치는건 감이 오질 않습니다 

  const isNameValid = (e) => {
    const isValid = e.target.value.length;
    setFirstNameValid(isValid);
    setUserNameValid(isValid);
  }; 
isFirstNameValid 와 isUserNameValid의 함수 내용이 같아서 위와같이 바꾸니 둘이 함께 바뀌네요

Reply:
@ByeongSooLee-KR input 태그에 name 속성을 주고 e 객체로 해당 name을 읽으면 안에서 각 input 태그별로 다른 동작을 할 수 있습니다

Reply:
요런 식인가요? 
  const isNameValid = (e) => {
    const isValid = e.target.value.length;
    e.target.name === "firstName"
      ? setFirstNameValid(isValid)
      : setUserNameValid(isValid);
  };